From 98f45927f555194da2107b06df1587edbcbfcb51 Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Tue, 10 Aug 2021 17:00:18 +0800
Subject: [PATCH] INTERNAL: Check height & width before create resource

We should confirm height & width valid before call
create_resource, otherwise virglrenderer driver will
run into error.

Tracked-On: OAM-97927
Signed-off-by: Ren Chenglei <chenglei.ren@intel.com>
---
 src/gallium/winsys/virgl/drm/virgl_drm_winsys.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c b/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c
index 7c111e9e7f5..f2336be6fa7 100644
--- a/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c
+++ b/src/gallium/winsys/virgl/drm/virgl_drm_winsys.c
@@ -292,6 +292,11 @@ virgl_drm_winsys_resource_create(struct virgl_winsys *qws,
    createcmd.stride = stride;
    createcmd.size = size;
 
+   if ((createcmd.height == 0) || (createcmd.width == 0)) {
+      FREE(res);
+      return NULL;
+   }
+
    ret = drmIoctl(qdws->fd, DRM_IOCTL_VIRTGPU_RESOURCE_CREATE, &createcmd);
    if (ret != 0) {
       FREE(res);
-- 
2.40.1

